tooling/internal/matlab/types/types.go

```diff
@@ -0,0 +1,1097 @@
+					})
+					for _, field := range rec.Fields {
+						fieldName := common.FieldIdentifierName(field.Name)
+						_, defaultExpressionKind := typeDefault(field.Type, rec.Namespace, "", st)
+						switch defaultExpressionKind {
+						case defaultValueKindNone:
+							fmt.Fprintf(w, "if ~isfield(kwargs, \"%s\")\n", fieldName)
+							common.WriteBlockBody(w, func() {
+								fmt.Fprintf(w, "throw(yardl.TypeError(\"Missing required keyword argument '%s'\"))\n", fieldName)
+							})
+						}
+						fmt.Fprintf(w, "self.%s = kwargs.%s;\n", fieldName, fieldName)
+					}
+				})
+				w.WriteStringln("")
+
+				// Computed Fields
+				if len(rec.ComputedFields) > 0 {
+					for _, computedField := range rec.ComputedFields {
+						fieldName := common.ComputedFieldIdentifierName(computedField.Name)
+
+						fmt.Fprintf(w, "function res = %s(self)\n", fieldName)
+						common.WriteBlockBody(w, func() {
+							common.WriteComment(w, computedField.Comment)
+							writeComputedFieldExpression(w, computedField.Expression, rec.Namespace)
+						})
+						w.WriteStringln("")
+					}
+					w.WriteStringln("")
+				}
+
+				// eq method
+				w.WriteStringln("function res = eq(self, other)")
+				common.WriteBlockBody(w, func() {
+					w.WriteStringln("res = ...")
+					w.Indented(func() {
+						fmt.Fprintf(w, "isa(other, \"%s\")", common.TypeSyntax(rec, rec.Namespace))
+						for _, field := range rec.Fields {
+							w.WriteStringln(" && ...")
+							fieldIdentifier := common.FieldIdentifierName(field.Name)
+							w.WriteString(typeEqualityExpression(field.Type, "self."+fieldIdentifier, "other."+fieldIdentifier))
+						}
+						w.WriteStringln(";")
+					})
+				})
+				w.WriteStringln("")
+
+				// neq method
+				w.WriteStringln("function res = ne(self, other)")
+				common.WriteBlockBody(w, func() {
+					w.WriteStringln("res = ~self.eq(other);")
+				})
+			})
+			w.WriteStringln("")
+
+			if !requiresArgsToConstruct {
+				w.WriteStringln("methods (Static)")
+				common.WriteBlockBody(w, func() {
+					writeZerosStaticMethod(w, common.TypeSyntax(rec, rec.Namespace), []string{})
+				})
+			}
+		})
+	})
+}
+
+func writeZerosStaticMethod(w *formatting.IndentedWriter, typeSyntax string, defaultArgs []string) {
+	// zeros method, only if can be constructed without arguments
+	w.WriteStringln("function z = zeros(varargin)")
+	common.WriteBlockBody(w, func() {
+		fmt.Fprintf(w, "elem = %s(%s);\n", typeSyntax, strings.Join(defaultArgs, ", "))
+		w.WriteStringln("if nargin == 0")
+		common.WriteBlockBody(w, func() {
+			w.WriteStringln("z = elem;")
+			w.WriteStringln("return;")
+		})
+
+		w.WriteStringln("sz = [varargin{:}];")
+		w.WriteStringln("if isscalar(sz)")
+		common.WriteBlockBody(w, func() {
+			w.WriteStringln("sz = [sz, sz];")
+		})
+		w.WriteStringln("z = reshape(repelem(elem, prod(sz)), sz);")
+	})
+}
+
+func typeEqualityExpression(t dsl.Type, a, b string) string {
+	if hasSimpleEquality(t) {
+		return fmt.Sprintf("all([%s] == [%s])", a, b)
```

Comment:
I'm wondering if this is the most efficient way of doing this, if a and b are (supposed to be), simple integers.

Reply:
Fixed.